src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -270,6 +321,95 @@ boolean checkContains(JsonElement expectedJson) {
         return pass;
     }
 
+    /**
+     * Checks the actual response json payload contains a key containing a JsonObject
+     * containing each of the pair values provided. The jsonKeys should be passed in
+     * as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonKeys      - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedPairs a hashmap with string key value pairs expected in the json
+     *                      response
+     */
+    @SuppressWarnings("squid:S1201")
+    abstract void contains(List<String> jsonKeys, Map<String, Object> expectedPairs);
+
+    /**
+     * Checks the actual response json payload contains a key containing a JsonObject
+     * containing each of the pair values provided. The jsonKeys should be passed in
+     * as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs    - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedPairs a hashmap with string key value pairs expected in the json
+     *                      response
+     */
+    boolean checkContains(List<String> jsonCrumbs, Map<String, Object> expectedPairs) {
+        JsonObject actualValue = this.response.getObjectData();
+        for (String jsonCrumb : jsonCrumbs) {
+            if (actualValue == null) {
+                break;
+            }
+            if (actualValue.get(jsonCrumb) instanceof JsonObject) {
+                actualValue = actualValue.get(jsonCrumb).getAsJsonObject();
+            } else {
+                actualValue = null;
+                break;
+            }
+        }
+        boolean pass = doesJsonObjectContainPair(expectedPairs, actualValue);
+        if (pass) {
+            this.reporter.pass("", EXPECTED_TO_FIND + STARTI + Reporter.formatHTML(String.join(" \uD83E\uDC1A ", jsonCrumbs)) + ENDI + " containing: " + DIV_I + Reporter.formatKeyPair(expectedPairs) + END_IDIV, FOUND + DIV_I + Reporter.formatHTML(GSON.toJson(actualValue)) + END_IDIV);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " containing: " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)